src/BenchmarksServer/Startup.cs

```diff
@@ -171,20 +172,34 @@ public static int Main(string[] args)
 
         private static string CloneAndRestore(string path, ServerJob job)
         {
-            var benchmarksDir = Git.Clone(path, _benchmarksRepoUrl);
+            // It's possible that the user specified a custom branch/commit for the benchmarks repo,
+            // so we need to add that to the set of sources to restore if it's not already there.
+            //
+            // Note that this is also going to de-dupe the repos if the same one was specified twice at
+            // the command-line (last first to support overrides).
+            var repos = new HashSet<Source>(job.Sources.Reverse(), SourceRepoComparer.Instance);
+            repos.Add(new Source() { Repository = _benchmarksRepoUrl });
```

Comment:
Funnily enough to be able to test changes locally you need to pass a custom commitish for aspnet/Benchmarks, which this didn't support.'

I'm realizing now that the `.Reverse()` probably belongs in the commandline driver code and not here. I'll remove it.